homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,179 @@
+"""Common code for decora_wifi."""
+
+from __future__ import annotations
+
+from decora_wifi import DecoraWiFiSession
+from decora_wifi.models.iot_switch import IotSwitch
+from decora_wifi.models.person import Person
+from decora_wifi.models.residence import Residence
+from decora_wifi.models.residential_account import ResidentialAccount
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, LIGHT_DOMAIN, PLATFORMS
+
+
+class DecoraWifiError(HomeAssistantError):
+    """Base for errors raised when the decora_wifi integration encounters an issue."""
+
+
+class SessionEntityNotFound(DecoraWifiError):
+    """Raised if a platform fails to find the session entity during setup."""
+
+
+class LoginFailed(DecoraWifiError):
+    """Raised when DecoraWifiPlatform.login() fails to log in."""
+
+
+class CommFailed(DecoraWifiError):
+    """Raised when DecoraWifiPlatform.login() fails to communicate with the myLeviton Service."""
+
+
+class DecoraWifiPlatform:
+    """Class to hold decora_wifi platform sessions and related methods."""
+
+    def __init__(self, email: str, password: str) -> None:
+        """Iniialize session holder."""
+        self._session = DecoraWiFiSession()
```

Comment:
We don't want side effects in `__init__` if at all possible.

Create this in `setup`